src/components/views/elements/Pill.tsx

```diff
@@ -79,6 +81,39 @@ export const Pill: React.FC<PillProps> = ({ type: propType, url, inMessage, room
     };
 
     const tip = hover && resourceId ? <Tooltip label={resourceId} alignment={Alignment.Right} /> : null;
+    let content: ReactElement | null = null;
+
+    switch (type) {
+        case PillType.AtRoomMention:
+        case PillType.RoomMention:
+            {
+                const avatar = targetRoom ? (
+                    <RoomAvatar room={targetRoom} width={16} height={16} aria-hidden="true" />
+                ) : null;
+                content = (
+                    <>
+                        {shouldShowPillAvatar && avatar}
+                        <span className="mx_Pill_linkText">{text}</span>
+                    </>
+                );
+            }
+            break;
+        case PillType.UserMention:
+            {
+                const avatar = <MemberAvatar member={member} width={16} height={16} aria-hidden="true" hideTitle />;
+                content = (
+                    <>
+                        {shouldShowPillAvatar && avatar}
+                        <span className="mx_Pill_linkText">{text}</span>
+                    </>
+                );
```

Comment:
this logic looks the same for both branches of the switch statement. Is there a reason we need to repeat it?